modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/IgnitePdsDestroyCacheTest.java

```diff
@@ -117,6 +128,135 @@ public void testDestroyCacheOperationNotBlockingCheckpointTest_LocalCache() thro
         doTestDestroyCacheOperationNotBlockingCheckpointTest(true);
     }
 
+    /**
+     * Tests cache destry with hudge dirty pages.
+     */
+    @Test
+    public void testDestroyCache() throws Exception {
+        IgniteEx ignite = startGrid(0);
+
+        ignite.cluster().active(true);
+
+        ignite.getOrCreateCache(new CacheConfiguration<>(DEFAULT_CACHE_NAME).setGroupName("gr1"));
+        ignite.getOrCreateCache(new CacheConfiguration<>("cache2").setGroupName("gr1"));
+
+        try (IgniteDataStreamer<Object, Object> streamer2 = ignite.dataStreamer(DEFAULT_CACHE_NAME)) {
+            PageMemoryEx pageMemory = (PageMemoryEx)ignite.cachex(DEFAULT_CACHE_NAME).context().dataRegion().pageMemory();
+
+            long totalPages = pageMemory.totalPages();
+
+            for (int i = 0; i <= totalPages; i++)
+                streamer2.addData(i, new byte[pageMemory.pageSize() / 2]);
+        }
+
+        ignite.destroyCache(DEFAULT_CACHE_NAME);
+    }
+
+    /**
+     * Tests partitioned cache destry with hudge dirty pages.
+     */
+    @Test
+    public void testDestroyCacheNotThrowsOOMPartitioned() throws Exception {
+        doTestDestroyCacheNotThrowsOOM(false);
+    }
+
+    /**
+     * Tests local cache destry with hudge dirty pages.
+     */
+    @Test
+    public void testDestroyCacheNotThrowsOOMLocal() throws Exception {
+        doTestDestroyCacheNotThrowsOOM(true);
+    }
+
+    /** */
+    public void doTestDestroyCacheNotThrowsOOM(boolean loc) throws Exception {
+        int batchSize = 1000;
+
+        int pageSize = 1024;
+
+        int partitions = 32;
+
+        DataStorageConfiguration ds = new DataStorageConfiguration()
+            .setDefaultDataRegionConfiguration(new DataRegionConfiguration()
+                .setMaxSize(batchSize * pageSize * partitions)
+                .setPersistenceEnabled(true))
+            .setPageSize(pageSize);
+
+        int payLoadSize = pageSize / 2;
+
+        IgniteConfiguration cfg = getConfiguration().setDataStorageConfiguration(ds);
+
+        final IgniteEx ignite = startGrid(cfg);
+
+        ignite.cluster().active(true);
+
+        startGroupCachesDynamically(ignite, loc);
+
+        GridCacheDatabaseSharedManager db =
+            (GridCacheDatabaseSharedManager)ignite.context().cache().context().database();
+
+        final CacheGroupContext grp = ignite.cachex(cacheName(0)).context().group();
+
+        final IgniteCacheOffheapManager offheap = grp.offheap();
+
+        Field batchField = U.findField(IgniteCacheOffheapManagerImpl.class, "BATCH_SIZE");
+
+        assertEquals(batchField.getInt(offheap), batchSize);
+
+        final AtomicLong progress = new AtomicLong();
+
+        PageMemoryEx pageMemory = (PageMemoryEx) ignite.cachex(cacheName(0)).context().dataRegion().pageMemory();
+
+        db.enableCheckpoints(false).get();
+
+        runAsync(() -> {
```

Comment:
Wait to completion this before test was stopped, other way you can get problem on kernel stop.

Reply:
i fix it through  fut.cancel()